util/Pos/Util/Util.hs

```diff
@@ -285,8 +285,15 @@ multilineBounds maxSize = F.later formatList
 logException :: LoggerName -> IO a -> IO a
 logException name = E.handleAsync (\e -> handler e >> E.throw e)
   where
+    -- Two ways of logging are used.
+    -- 1. log-warper is used, because it can log to files, etc.
+    -- 2. Simple 'putStrLn' is used, because something may be wrong
+    -- with log-warper and this function is useful in exceptional situations.
     handler :: E.SomeException -> IO ()
-    handler = usingLoggerName name . logError . pretty
+    handler exc = do
```

Comment:
On MacOS, this will work, but it won't work on Windows since we silence all `stdout` and `stderr` there. To work around that, see `reportErrorDefault` [here](https://github.com/input-output-hk/cardano-sl/blob/604f58270f790a2338d854ea23214640a4b9baba/tools/src/launcher/Main.hs#L174) and consider either using it or rewriting it to avoid duplicating entities.

Reply:
stdout is visible on windows, if the program is manually ran from a terminal, and doesnt go out of its way to silence its own stdout

Reply:
@cleverca22 are you absolutely sure about this? I remember having to write this function because there was no `stdout` output on Windows, but I may be mistaken.

Reply:
```
module Main where

import System.IO

main :: IO ()
main = do
    hPutStrLn stdout "hello world"
    hPutStrLn stderr "and std err"
```

```
C:\stack>stack ghc -- example.hs -o example.exe
[1 of 1] Compiling Main             ( example.hs, example.o )
Linking example.exe ...

C:\stack>example.exe
hello world
and std err
```

Reply:
https://github.com/input-output-hk/cardano-sl/blob/develop/tools/src/launcher/Main.hs#L296-L301

Reply:
@cleverca22 yes, those lines are exactly what I mean by "silencing all `stdout` and `stderr`". Can you test this code?

```
module Main where

import System.IO
import qualified System.IO.Silently as Silently

main :: IO ()
main =
    Silently.hSilence [stdout, stderr] $ do
        hPutStrLn stdout "hello world"
        hPutStrLn stderr "and std err"
```

Reply:
cant get that one to compile under windows, since i dont have nix helping out and auto-loading all of hackage

but it does make sense why that would fail, i'm thinking that part of launcher should either be removed or made optional

why was that added originally?

Reply:
@cleverca22 there is a comment in the lines you linked to [here](https://github.com/input-output-hk/cardano-sl/pull/2421#discussion_r165798894) explaining that.

Reply:
> On MacOS, this will work, but it won't work on Windows since we silence all stdout and stderr there.

What exactly won't work on Windows? `putStrLn` will do nothing in `cardano-launcher`, that's true. But `logError` will log `message` to a file. That's exactly what we want on Windows, isn't it? For `cardano-node` and other executables we don't silence stdout/stderr, so it's not different from MacOS at all.

> To work around that, see reportErrorDefault here and consider either using it directly

Those are some cardano-launcher-specific hacks, I don't want to have them in `cardano-sl-util`.

> but please avoid duplicating entities.

That's the only problem, but I think it's minor. If you strictly don't like it, I will try to change it. I think the best way is to temporary disable terminal logging for logger name `name` (only for the `logError` call below). I've asked in #log-warper channel how to do it.

Reply:
> -- This function is only intended to be used before normal logging
> -- is initialized.

In this PR `logException` is used after logging is initialized. Which is fine for CSL-2064, because there we definitely do initialize logging successfully (strange things happen much later).

Reply:
@Pastafarianist okay, I changed this function, please re-review.